mllib/src/main/scala/org/apache/spark/ml/clustering/KMeans.scala

```diff
@@ -306,12 +307,20 @@ class KMeans @Since("1.5.0") (
 
   @Since("2.0.0")
   override def fit(dataset: Dataset[_]): KMeansModel = {
+    val handlePersistence = dataset.rdd.getStorageLevel == StorageLevel.NONE
+    fit(dataset, handlePersistence)
+  }
+
+  @Since("2.2.0")
+  protected def fit(dataset: Dataset[_], handlePersistence: Boolean): KMeansModel = {
```

Comment:
Hi @jkbradley, 
In actually, I thought of the same thing, but Mr. yuhao yang advised me to do the same style of LogisticRegression.scala
anywhere I changed the file as you want, and I changed the code a little (e.g : no need to generate the rdd two times
the first one to check storage level and the second one for Kmeans
)
here the change
https://github.com/ZakariaHili/spark/commit/58de5499d0930e4019c94718a2a9ed3dd715acb2

do you still want from me to send a PR ?